OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+            Assert.AreEqual(expectedValue, variableValue);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.INFO,
+                $@"Returning variable value ""{variableValue}"" for variation ""{variation.Key}"" of feature flag ""{featureKey}""."));
+        }
+
+        #endregion // Test GetFeatureVariableValueForType method
+
+        #region Test IsFeatureEnabled method
+
+        // Should return false and log error message when arguments are null or empty.
+        [Test]
+        public void TestIsFeatureEnabledGivenNullOrEmptyArguments()
+        {
+            var featureKey = "featureKey";
+
+            Assert.False(Optimizely.IsFeatureEnabled(featureKey, null, null));
+            Assert.False(Optimizely.IsFeatureEnabled(featureKey, "", null));
+
+            Assert.False(Optimizely.IsFeatureEnabled(null, TestUserId, null));
+            Assert.False(Optimizely.IsFeatureEnabled("", TestUserId, null));
+
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "User ID must not be empty."), Times.Exactly(2));
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "Feature flag key must not be empty."), Times.Exactly(2));
+        }
+
+        // Should return false and log error message when feature flag key is invalid.
+        [Test]
+        public void TestIsFeatureEnabledGivenFeatureFlagNotFound()
+        {
+            var featureKey = "this_feature_should_never_be_found_in_the_datafile_unless_the_datafile_creator_got_insane";
+            Assert.False(Optimizely.IsFeatureEnabled(featureKey, TestUserId, null));
+            
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, $@"Feature key ""{featureKey}"" is not in datafile."));
+        }
+
+        // Should return false and log error message when arguments are null or empty.
+        [Test]
+        public void TestIsFeatureEnabledGivenFeatureFlagContainsInvalidExperiment()
+        {
+            var tempConfig = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, new NoOpErrorHandler());
+            var featureFlag = tempConfig.GetFeatureFlagFromKey("multi_variate_feature");
+
+            var optly = Helper.CreatePrivateOptimizely();
+            optly.SetFieldOrProperty("Config", tempConfig);
+
+            // Set such an experiment to the list of experiment ids, that does not belong to the feature.
+            featureFlag.ExperimentIds = new List<string> { "4209211" };
+
+            // Should return false when the experiment in feature flag does not get found in the datafile.
+            Assert.False((bool)optly.Invoke("IsFeatureEnabled", "multi_variate_feature", TestUserId, null));
+
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, @"Experiment ID ""4209211"" is not in datafile."));
+        }
+
+        // Should return default value and log message when feature is not enabled for the user.
```

Comment:
is feature enabled does not return default value

Reply:
Corrected.